crates/stackable-webhook/src/servers/conversion.rs

```diff
@@ -1,14 +1,61 @@
 use std::fmt::Debug;
 
-use axum::{Json, Router, extract::State, routing::post};
+use axum::{Json, Router, routing::post};
+use k8s_openapi::{
+    ByteString,
+    apiextensions_apiserver::pkg::apis::apiextensions::v1::{
+        CustomResourceConversion, CustomResourceDefinition, ServiceReference, WebhookClientConfig,
+        WebhookConversion,
+    },
+};
 // Re-export this type because users of the conversion webhook server require
 // this type to write the handler function. Instead of importing this type from
 // kube directly, consumers can use this type instead. This also eliminates
 // keeping the kube dependency version in sync between here and the operator.
 pub use kube::core::conversion::ConversionReview;
+use kube::{
+    Api, Client, ResourceExt,
+    api::{Patch, PatchParams},
+};
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::cli::OperatorEnvironmentOptions;
+use tokio::{sync::mpsc, try_join};
 use tracing::instrument;
+use x509_cert::{
+    Certificate,
+    der::{EncodePem, pem::LineEnding},
+};
 
-use crate::{StatefulWebhookHandler, WebhookHandler, WebhookServer, options::Options};
+use crate::{
+    WebhookError, WebhookHandler, WebhookServer, constants::DEFAULT_HTTPS_PORT, options::Options,
+};
+
+#[derive(Debug, Snafu)]
+pub enum ConversionWebhookError {
+    #[snafu(display("failed to create webhook server"))]
+    CreateWebhookServer { source: WebhookError },
+
+    #[snafu(display("failed to run webhook server"))]
+    RunWebhookServer { source: WebhookError },
+
+    #[snafu(display("failed to receive certificate from channel"))]
+    ReceiverCertificateFromChannel,
+
+    #[snafu(display("failed to convert CA certificate into PEM format"))]
+    ConvertCaToPem { source: x509_cert::der::Error },
+
+    #[snafu(display("failed to reconcile CRDs"))]
+    ReconcileCRDs {
+        #[snafu(source(from(ConversionWebhookError, Box::new)))]
+        source: Box<ConversionWebhookError>,
+    },
+
+    #[snafu(display("failed to update CRD {crd_name:?}"))]
+    UpdateCRD {
```

Comment:
note: I think we should stick with PascalCase here.


```suggestion
    ReconcileCrd {
        #[snafu(source(from(ConversionWebhookError, Box::new)))]
        source: Box<ConversionWebhookError>,
    },

    #[snafu(display("failed to update CRD {crd_name:?}"))]
    UpdateCrd {
```

Reply:
fe1749d